cmd/nodepool/create.go

```diff
@@ -121,6 +121,27 @@ func (o *CreateNodePoolOptions) Run(ctx context.Context) error {
 
 	switch nodePool.Spec.Platform.Type {
 	case hyperv1.AWSPlatform:
+		if len(o.InstanceProfile) == 0 {
+			o.InstanceProfile = fmt.Sprintf("%s-worker", hcluster.Spec.InfraID)
+		}
+		if len(o.SubnetID) == 0 {
+			if hcluster.Spec.Platform.AWS.CloudProviderConfig.Subnet.ID != nil {
+				o.SubnetID = *hcluster.Spec.Platform.AWS.CloudProviderConfig.Subnet.ID
+			} else {
+				return fmt.Errorf("subnet ID was not specified and cannot be determined from HostedCluster")
+			}
+		}
+		if len(o.SecurityGroupID) == 0 {
+			defaultNodePool := &hyperv1.NodePool{}
+			if err := client.Get(ctx, types.NamespacedName{Namespace: hcluster.Namespace, Name: hcluster.Name}, defaultNodePool); err != nil {
+				return fmt.Errorf("security group ID was not specified and cannot be determined from default nodepool: %v", err)
```

Comment:
This will make the command fail if the default nodePool does not exists. We might want to be more flexible and not fail if the err is not found here?

Reply:
The error is only thrown if you have not specified a security group id. So if you didn't specify one and we can't find the default nodepool to copy it from, then your nodepool will be broken anyway. Might as well fail early.